src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,44 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private static final BulldozerProducer BULLDOZER_PRODUCER = new BulldozerProducer();
+    private static final ExcavatorProducer EXCAVATOR_PRODUCER = new ExcavatorProducer();
+    private static final TruckProducer TRUCK_PRODUCER = new TruckProducer();
+
+    @Override
+    public List<Machine> getAll(Class type) {
```

Comment:
param type should be parameterized
```suggestion
    public List<Machine> getAll(Class<? extends Machine> type) {
```